src/cursor_ext.rs

```diff
@@ -53,10 +53,22 @@ impl CursorExt for Cursor<Vec<u8>> {
     }
 
     fn write_string(&mut self, v: &str) -> Result<(), Error> {
-        let len = v.len() + 1;
-        self.write_i32::<LittleEndian>(len as i32)?;
-        let _ = self.write(v.as_bytes())?;
-        let _ = self.write(&[0u8; 1])?;
+        if v.is_ascii() {
+            // ASCII strings do not require encoding
+            let len = v.len() + 1;
+            self.write_i32::<LittleEndian>(len as i32)?;
+            let _ = self.write(v.as_bytes())?;
```

Comment:
I am new to Rust so I could be missing something here, and feel free to educate me if that is the case. However, I see no material difference between `write` and `write_all` for Vec based on what I see in https://github.com/rust-lang/rust/blob/15e8a67/src/libstd/io/impls.rs#L211-L222

Either way, it may be best to open a new issue to discuss this, or a new PR to fix it, since this feedback is about code that is already in main. I did not modify this code, I've just moved it inside an if block.

Reply:
Yeah this could cause problems, however as it was already in the repo before, it's not going to block this PR, this should be changed at some point though.